godot-macros/src/docs.rs

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+mod markdown_converter;
+
+use crate::class::{ConstDefinition, Field, FuncDefinition, SignalDefinition};
+use proc_macro2::{Ident, TokenStream};
+use quote::{quote, ToTokens};
+use venial::*;
+
+pub fn make_definition_docs(
+    base: String,
+    description: &[Attribute],
+    members: &[Field],
+) -> TokenStream {
+    (|| {
+        let desc = make_docs_from_attributes(description)?;
+        let members = members
+            .into_iter()
+            .filter(|x| x.var.is_some() | x.export.is_some())
+            .map(member)
+            .collect::<Option<String>>()?;
+        Some(quote! {
+            docs: ::godot::docs::StructDocs {
+                base: #base,
+                description: #desc,
+                members: #members,
+            }.into()
+        })
+    })()
+    .unwrap_or(quote! { docs: None })
+}
+
+pub fn make_inherent_impl_docs(
+    functions: &[FuncDefinition],
+    constants: &[ConstDefinition],
+    signals: &[SignalDefinition],
+) -> TokenStream {
+    fn pieces(
+        functions: &[FuncDefinition],
+        signals: &[SignalDefinition],
+        constants: &[ConstDefinition],
+    ) -> Option<TokenStream> {
+        let methods = functions
+            .iter()
+            .map(make_method_docs)
+            .collect::<Option<String>>()?;
+        let signals = signals
+            .iter()
+            .map(make_signal_docs)
+            .collect::<Option<String>>()?;
+        let constants = constants
+            .iter()
+            .map(|ConstDefinition { raw_constant: x }| x)
+            .map(make_constant_docs)
+            .collect::<Option<String>>()?;
+        let field_definition = quote! {
+            docs: ::godot::docs::InherentImplDocs {
+                methods: #methods,
+                signals: #signals,
+                constants: #constants,
+            }.into()
+        };
+        Some(field_definition)
+    }
+    pieces(functions, signals, constants).unwrap_or(quote! { docs: None })
```

Comment:
```suggestion
    pieces(functions, signals, constants).unwrap_or_else(|| quote! { docs: None })
```
to evaluate lazily

Reply:
but quote! is a proc macro which is evaluated at compile time, no?

Reply:
Yes, but it creates a `TokenStream` which is then allocated at runtime.